pipeline/pipeline.go

```diff
@@ -321,9 +345,11 @@ func (p *Pipeline) In(sourceID SourceID, sourceName string, offset int64, bytes
 	event.streamName = DefaultStreamName
 	event.Size = len(bytes)
 
+	p.inSampleMu.Lock()
```

Comment:
We can't use lock here as it super hot path. Let's think about how to avoid mutex. Maybe cache length to atomic var and use lock only inside the condition.